### DISCUSSION TITLE
Multiple schema files with query / mutation types and codegen

### DISCUSSION THREAD

**Discussion by tammomueller** [Netflix/dgs-framework#688] [General]
Hello,

I want to follow a schema-first workflow and have a large number of domain specific types, queries and mutations. I'd like to break them out into domain specific schema files, however, defining the query / mutation type causes duplication issues during code gen, 

Any help is appreciated.

**Comment by srinivasankavitha** [comment]
you can use the `extend` keyword to define Query and Mutation  extension definitions.
`extend type Query  {
 // your type
}

**Reply by tammomueller** [reply]
Thanks
